sinks/influxdb/driver.go

```diff
@@ -85,101 +94,83 @@ func (self *influxdbSink) metricToSeries(timeseries *sink_api.Timeseries) *influ
 	if timeseries.MetricDescriptor.Type.String() != "" {
 		seriesName = fmt.Sprintf("%s_%s", seriesName, timeseries.MetricDescriptor.Type.String())
 	}
-
-	// Add the real metric value.
-	columns = append(columns, "value")
-	values = append(values, timeseries.Point.Value)
+	point := influxdb.Point{
+		Measurement: seriesName,
+		Tags:        make(map[string]string, 0, len(timeseries.Point.Labels)),
+		Fields: map[string]interface{}{
+			"value": timeseries.Point.Value,
+		},
+		Time: timeseries.Point.End.UTC(),
+	}
 	// Append labels.
-	if !self.c.avoidColumns {
-		for key, value := range timeseries.Point.Labels {
-			columns = append(columns, key)
-			values = append(values, value)
+	for key, value := range timeseries.Point.Labels {
+		if value != "" {
+			point.Tags[key] = value
 		}
-	} else {
-		seriesName = strings.Replace(seriesName, "/", "_", -1)
-		seriesName = fmt.Sprintf("%s_%s", util.LabelsToString(timeseries.Point.Labels, "_"), seriesName)
 	}
-	// Add timestamp.
-	columns = append(columns, "time")
-	values = append(values, timeseries.Point.End.Unix())
-	// Ass sequence number
-	columns = append(columns, "sequence_number")
-	values = append(values, self.seqNum.Get(seriesName))
 
-	return self.newSeries(seriesName, columns, values)
-}
-
-var eventColumns = []string{
-	"time",                     // Column 0
-	"sequence_number",          // Column 1
-	sink_api.LabelPodId.Key,    // Column 2
-	sink_api.LabelPodName.Key,  // Column 3
-	sink_api.LabelHostname.Key, // Column 4
-	"value",                    // Column 5
+	return point
 }
 
 // Stores events into the backend.
 func (sink *influxdbSink) StoreEvents(events []kube_api.Event) error {
-	dataPoints := []*influxdb.Series{}
+	if err := sink.createDatabase(); err != nil {
```

Comment:
If InfluxDB is not up before heapster starts, db creation will fail in Register. To avoid the race condition and to avoid db being accidentally deleted while heapster is running, heapster tries to re-create db is required.